src/registry/artifactory.rs

```diff
@@ -58,10 +83,10 @@ impl Artifactory {
     /// Pings artifactory to ensure registry access is working
     pub async fn ping(&self) -> miette::Result<()> {
         let repositories_url: Url = {
-            let mut uri = self.registry.to_owned();
+            let mut uri: url::Url = self.registry.clone();
```

Comment:
See above comment for the enum approach

Reply:
52fe4ef5f2fddf2cba769cfe296609f7ec27800a